.yamato/yamato-config.yml

```diff
@@ -10,12 +10,19 @@ commands:
     - python3 -m pip install unity-downloader-cli --index-url https://artifactory.prd.it.unity3d.com/artifactory/api/pypi/pypi/simple --upgrade
     - unity-downloader-cli -u 2020.2.0b9 -c editor -c StandaloneSupport-IL2CPP -c Linux --wait --published
     - git clone git@github.cds.internal.unity3d.com:unity/utr.git utr
-    - utr/utr --testproject=./TestUrdfImporter --editor-location=.Editor --reruncount=0 --artifacts_path=test-results --suite=playmode --suite=editor --platform=Editor
+    - utr/utr --testproject=./TestUrdfImporter --editor-location=.Editor --reruncount=0 --artifacts_path=test-results --suite=playmode --suite=editor --platform=Editor --enable-code-coverage --coverage-results-path=../test-results --coverage-options="assemblyFilters:+Unity.Robotics.URDFImporter,+Unity.Robotics.URDFImporter.Editor;generateHtmlReport;generateBadgeReport;generateAdditionalMetrics"
+    # check test coverage
+    - command: |
+        linecoverage=$(cat test-results/Report/Summary.xml | grep Linecoverage | grep -Eo '[+-]?[0-9]+([.][0-9]+)?')
+        echo "Line coverage: $linecoverage%"
+        if [[ $linecoverage < 0 ]]; then exit 1; fi
 triggers:
     cancel_old_ci: true
     expression: |
         (pull_request.target eq "main"  AND
         NOT pull_request.push.changes.all match "**/*.md") OR
+        (pull_request.target eq "dev"  AND
+        NOT pull_request.push.changes.all match "**/*.md") OR
         (push.branch eq "dev" AND
         NOT push.changes.all match "**/*.md")
```

Comment:
We should keep one or the other of these two blocks or we'll end up running redundant tests. There were two reasons I used the original `push.branch eq "dev"` logic instead of `pull_request.target eq "dev"`
1. `pull_request.target` will trigger a yamato run not just when a PR is submitted, but on every subsequent update to that PR too, which can cause a lot of churn occasionally. But, maybe the threat of churn is a good forcing function to encourage devs to run these tests locally before submitting their PR...
2. Yamato has a tendency to become unstable and get backed up, which can make testing times go from ~5 minutes to 45+ minutes. If we block merges pending Yamato results, we'll definitely run into times where a PR is approved, but unable to be merged for an hour because of Yamato issues... We could mitigate this, though, by allowing manual merge overrides in  cases where Yamato is acting extra slow and we want to get something in faster.
This is totally up for discussion, I don't have a strong opinion either way. I set it up the way I did originally to reduce friction, but there's definitely a lot of benefits to forcing PRs to wait for test results.

Reply:
Ah, thanks, Devin! I didn't know all these potential issues on Yamato. It makes sense to me that we shouldn't be blocked by the Yamato issue. I personally often forget to rerun the unit tests after dealing with PR comments, so I thought it might be a concern that the unit tests will not pass if we don't force the checking and it will be a huge trouble for the next committer.

Do you think it would be better if we can manually trigger the Yamato jobs right before merging them into the dev branch? (I am not sure if Yamato can do this or not though)

Reply:
> Do you think it would be better if we can manually trigger the Yamato jobs right before merging them into the dev branch? (I am not sure if Yamato can do this or not though)  
  
Yamato can do this, but I think we should think of anything that requires manual action to be not reliable enough for consideration.  In the case where we want tests to run _before_ merge, `cancel_old_ci: true` guarantees that we won't have to wait on stale tests. If a PR revision is submitted before the old tests are finished, the currently running tests are discarded and a new CI run is initiated with the fresh revision.

Reply:
Okay, get you! that makes sense.